packages/plugin-connector/src/data-sources/schemas/query-runner.schema.js

```diff
@@ -1,15 +1,90 @@
 import { z } from 'zod';
 
-export const QueryResultSchema = z.object({
-	rows: z.array(z.record(z.string(), z.any())),
-	columnTypes: z.array(
-		z.object({
-			name: z.string(),
-			evidenceType: z.enum(['boolean', 'number', 'string', 'date']),
-			typeFidelity: z.union([z.literal('precise'), z.literal('inferred')])
-		})
+export const QueryResultSchema = z
+	.object({
+		// Note that this only validates that the first item in the array
+		// is a record with string keys. If the connector returns some
+		// inconsistent array (e.g. [{}, 1]), it will not detect the
+		// invalid row.
+		rows: z.array(z.any()).refine(
+			(data) => {
+				// result is not an array, fail
+				if (!Array.isArray(data)) return false;
+				// result has no rows, we can't validate this
+				// but this is a correct result set
+				if (data.length === 0) return true;
+				try {
+					z.record(z.string(), z.any()).parse(data[0]);
+				} catch {
+					return false;
+				}
+				return true;
+			},
+			{ message: 'Data connector returned invalid rows' }
+		),
+		columnTypes: z.array(
+			z.object({
+				name: z.string(),
+				evidenceType: z.enum(['boolean', 'number', 'string', 'date']),
+				typeFidelity: z.union([z.literal('precise'), z.literal('inferred')])
```

Comment:
There's also the method `.nativeEnum`

Reply:
We can make that change if/when we clean up the types in `db-commons`, for now I don't think that value isn't really used so we just want to make sure that it matches the expected interface until we change it